### ISSUE TITLE
CTE with RECURSIVE column does not exist

### ISSUE THREAD

**Issue by johnatannvmd** [kyleconroy/sqlc#723]
sqlc.yaml
```yaml
version: "1"
packages:
  - name: main
    path: .
    schema: schema.sql
    queries: query.sql
    engine: postgresql
    emit_json_tags: true
    emit_prepared_queries: false
    emit_interface: true
    emit_exact_table_names: false
    emit_empty_slices: false
```

schema.sql
```sql
create table graph
(
	id serial not null primary key,
	data int not null
);

create unique index graph_id_uindex
	on graph (id);

--

create table graph_link
(
	g_id int not null
		constraint graph_link_graph_id_fk
			references graph (id),
	import_g int not null
		constraint graph_link_graph_id_fk_2
			references graph (id)
);

create unique index graph_link_g_id_import_g_uindex
	on graph_link (g_id, import_g);
```
query.sql
```sql
-- name: GetGraphDeps :many
WITH RECURSIVE search_graph(id, import_id, data, depth, path, cycle) AS (
    SELECT gl.g_id id, gl.import_g import_id, g.data, 1 as depth,
           ARRAY[gl.g_id] as path, false as cycle
    FROM graph_link gl
    JOIN graph g on g.id = gl.g_id

    UNION ALL

    SELECT gl.g_id id, gl.import_g import_id, g.data, search_graph.depth + 1 as depth,
           search_graph.path || gl.import_g as path, gl.g_id = ANY(search_graph.path) as cycle
    FROM (graph_link gl JOIN graph g on g.id = gl.g_id),
         search_graph
    WHERE gl.g_id = search_graph.import_id AND NOT search_graph.cycle
)
SELECT DISTINCT search_graph.id, search_graph.depth, search_graph.data FROM search_graph;
```

`sqlc generate` gives me
```
query.sql:11:17: column "id" does not exist
```

**Comment by johnatannvmd** [comment]
@kindermoumoute playing with aliases names didn't help. I saw #709 issue before posting this.

**Comment by kindermoumoute** [comment]
It didn't help for me too, I have no idea why it worked for @jschaf 🤷‍♂️

**Comment by jschaf** [comment]
I wasn't doing a recursive CTE.

**Comment by kindermoumoute** [comment]
The below example will result in an error `column "modified_at" does not exist` similar to what you have.

```SQL
CREATE TABLE foo (
    key uuid DEFAULT public.uuid_generate_v4() NOT NULL,
    modification_date timestamp with time zone DEFAULT now() NOT NULL
);
```

```SQL
-- name: FooBar :many
WITH agg1 AS (
    SELECT t.key, t.modification_date as modified_at FROM foo t
)
SELECT key FROM agg1 WHERE modified_at > @from_date;
```

All aliases I tried failed but the original name works:

```SQL
-- name: FooBar :many
WITH agg1 AS (
    SELECT t.key, t.modification_date as modification_date FROM foo t
)
SELECT key FROM agg1 WHERE modification_date > @from_date;
```

Error comes from [this line](https://github.com/kyleconroy/sqlc/blob/b97f385c18e8f05fee8913341b71e81444919781/internal/compiler/resolve.go#L163-L169) so I would suggest the sqlc code is not indexing aliases from the WITH. Maybe @kyleconroy has some additional inputs on this?

**Comment by jschaf** [comment]
Maybe try a fully qualified reference or using a table alias? I have a pretty complex query with 5 CTEs that works and I'm using table aliases for each CTE. In general, I've found being painfully explicit with aliases and casting sometimes helps sqlc work through the query.

I also use this syntax for params: `sqlc.arg('EndTime')`, though it probably doesn't matter your case.

**Comment by kyleconroy** [comment]
Fixed in #896
